colcon_ros/package_identification/ros.py

```diff
@@ -110,15 +111,16 @@ def identify(self, desc):  # noqa: D102
             for _ in (1, ):
                 # try to get information from setup.cfg file
                 if setup_cfg.is_file():
-                    config = get_configuration(setup_cfg)
-                    name = config.get('metadata', {}).get('name')
-                    if name:
-                        options = config.get('options', {})
-
-                        def getter(env):
-                            nonlocal options
-                            return options
-                        break
+                    if is_reading_cfg_sufficient(setup_py):
```

Comment:
In the case where the setup.cfg is *not* sufficient, should we execute the else-block at L124?

Reply:
Yes, that is what is happening since the `for` loop was not `break`ed.

Reply:
:+1:  Missed that, thanks.